File: android/src/main/java/fr/wavyapp/reactNativeStarPrinter/Characters.kt
package fr.wavyapp.reactNativeStarPrinter

import com.starmicronics.stario10.starxpandcommand.display.InternationalCharacterType as DisplayInternationalCharacterType
import com.starmicronics.stario10.starxpandcommand.printer.InternationalCharacterType as PrinterInternationalCharacterType

fun printerInternationalCharacterTypeFromString(value: String? = "usa"): PrinterInternationalCharacterType {
  return when (value) {
    "usa" -> PrinterInternationalCharacterType.Usa
    "france" -> PrinterInternationalCharacterType.France
    "germany" -> PrinterInternationalCharacterType.Germany
    "uk" -> PrinterInternationalCharacterType.UK
    "denmark" -> PrinterInternationalCharacterType.Denmark
    "sweden" -> PrinterInternationalCharacterType.Sweden
    "italy" -> PrinterInternationalCharacterType.Italy
    "spain" -> PrinterInternationalCharacterType.Spain
    "japan" -> PrinterInternationalCharacterType.Japan
    "norway" -> PrinterInternationalCharacterType.Norway
    "denmark2" -> PrinterInternationalCharacterType.Denmark2
    "spain2" -> PrinterInternationalCharacterType.Spain2
    "latinAmerica" -> PrinterInternationalCharacterType.LatinAmerica
    "korea" -> PrinterInternationalCharacterType.Korea
    "ireland" -> PrinterInternationalCharacterType.Ireland
    "slovenia" -> PrinterInternationalCharacterType.Slovenia
    "croatia" -> PrinterInternationalCharacterType.Croatia
    "china" -> PrinterInternationalCharacterType.China
    "vietnam" -> PrinterInternationalCharacterType.Vietnam
    "arabic" -> PrinterInternationalCharacterType.Arabic
    "legal" -> PrinterInternationalCharacterType.Legal
    else -> PrinterInternationalCharacterType.Usa
  }
}

fun displayInternationalCharacterTypeFromString(value: String? = "usa"): DisplayInternationalCharacterType {
  return when (value) {
    "usa" -> DisplayInternationalCharacterType.Usa
    "france" -> DisplayInternationalCharacterType.France
    "germany" -> DisplayInternationalCharacterType.Germany
    "uk" -> DisplayInternationalCharacterType.UK
    "denmark" -> DisplayInternationalCharacterType.Denmark
    "sweden" -> DisplayInternationalCharacterType.Sweden
    "italy" -> DisplayInternationalCharacterType.Italy
    "spain" -> DisplayInternationalCharacterType.Spain
    "japan" -> DisplayInternationalCharacterType.Japan
    "norway" -> DisplayInternationalCharacterType.Norway
    "denmark2" -> DisplayInternationalCharacterType.Denmark2
    "spain2" -> DisplayInternationalCharacterType.Spain2
    "latinAmerica" -> DisplayInternationalCharacterType.LatinAmerica
    "korea" -> DisplayInternationalCharacterType.Korea
    else -> DisplayInternationalCharacterType.Usa
  }
}

File: ios/ReactNativeStarPrinter/Characters.swift
import StarIO10

func getPrinterInternationalCharacterType(name: String) -> StarXpandCommand.Printer.InternationalCharacterType {
  switch name {
    case "usa":
      return .usa
    case "france":
      return .france
    case "germany":
      return .germany
    case "uk":
      return .uk
    case "denmark":
      return .denmark
    case "sweden":
      return .sweden
    case "italy":
      return .italy
    case "spain":
      return .spain
    case "japan":
      return .japan
    case "norway":
      return .norway
    case "denmark2":
      return .denmark2
    case "spain2":
      return .spain2
    case "latinAmerica":
      return .latinAmerica
    case "korea":
      return .korea
    case "ireland":
      return .ireland
    case "slovenia":
      return .slovenia
    case "croatia":
      return .croatia
    case "china":
      return .china
    case "vietnam":
      return .vietnam
    case "arabic":
      return .arabic
    case "legal":
      return .legal
    default:
      return .usa
  }
}
